src/utils/popups.js

```diff
@@ -55,28 +58,70 @@ const wcLinkPopupHTML = (title, displayProps) => {
           ${downstreamWCLinks}`;
 };
 
+const fitMapToFeatures = (map, features) => {
+  var bounds = new mapboxgl.LngLatBounds();
+
+  features.forEach((feature) => {
+    if (feature.geometry.type === "LineString") {
+      feature.geometry.coordinates.forEach((coords) => {
+        bounds.extend(coords);
+      });
+    } else if (feature.geometry.type === "MultiLineString") {
+      feature.geometry.coordinates.forEach((line) => {
+        line.forEach((coords) => {
+          bounds.extend(coords);
+        });
+      });
+    }
+  });
+
+  map.fitBounds(bounds, { padding: 50 });
+};
+
 const getUpstream = async (id) => {
-  const url = `${waterNetworkAPIBase}/collections/WatercourseLink/items/${id}/upstream`;
+  const url = `${waterNetworkAPIBase}/collections/WatercourseLink/items/${id}/upstream?limit=10`;
```

Comment:
Goes with this [API PR](https://github.com/Swirrl/defra-water-network-api/pull/129)  to get fewer wclinks so the bbox isn't ridiculously huge when we query for all the wclinks, hydro nodes and sites